tests/mock_libuvc.c

```
@@ -0,0 +1,825 @@
+    usleep(2000); /* ~500 fps ceiling; consumption paces the real rate */
+  }
+  return NULL;
+}
+
+/* -------------------------------------------------------------------------- */
+/* libuvc API surface used by the element.                                    */
+/* -------------------------------------------------------------------------- */
+
+static uvc_device_t *mock_new_device(uvc_context_t *ctx, int index) {
+  uvc_device_t *dev = calloc(1, sizeof(*dev));
+  dev->ctx = ctx;
```

Cppcheck (2.20.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)